amazonka/src/Control/Monad/Error/AWS.hs

```diff
@@ -69,11 +69,12 @@ runAWST :: (MonadCatch m, MonadResource m, AWSEnv r)
 runAWST e = runExceptT . AWST.runAWST e
 {-# DEPRECATED runAWST "Exists for backwards compatibility with pre-@1.0@." #-}
 
-hoistError :: (MonadError e m, AWSError e) => Either Error a -> m a
+hoistError :: (Applicative m, MonadError e m, AWSError e) => Either Error a -> m a
 hoistError = either (throwing _Error) pure
```

Comment:
Would changing the use of `pure` to `return` remove the need for the `Applicative` constraint entirely?